models/repo_permission.go

```diff
@@ -102,6 +104,35 @@ func (p *Permission) CanWriteIssuesOrPulls(isPull bool) bool {
 	return p.CanWrite(unit.TypeIssues)
 }
 
+// CanWriteToBranch checks if the branch is writable by the user
+func (p *Permission) CanWriteToBranch(branch string) bool {
+	if p.CanWrite(unit.TypeCode) {
+		return p.CanWrite(unit.TypeCode)
+	}
+
+	prs, err := GetUnmergedPullRequestsByHeadInfo(p.Units[0].RepoID, branch)
```

Comment:
`p.Units[0].RepoID`


I'm pretty sure there are situations whereby the `Units` field is empty. I think a check can be added `len(p.Units) > 0`